packages/analyticsdx-template-lint/src/utils.ts

```diff
@@ -245,14 +245,14 @@ export function fuzzySearcher(
 /** Create a function that will cache the result of the underlying function on the first call, and
  * return that result from there out.
  */
-export function caching<A extends any[], R, T>(fn: (this: T, ...arg: A) => R): (this: T, ...arg: A) => R {
+export function caching<A extends any[], R>(fn: (...arg: A) => R): (...arg: A) => R {
   let result: R;
   let resultError: unknown | undefined;
-  let _fn: ((this: T, ...args: A) => R) | undefined = fn;
-  return function (this: T, ...args: A) {
```

Comment:
tslint was spitting out an error (but not failing) about on this return of a function rather than a lambda. We don't need the `this` handling here anyways, so I just switched it to a lambda.